src/main/java/br/com/sbk/sbking/core/HandEvaluations.java

```diff
@@ -90,4 +92,67 @@ public boolean isBalanced() {
         && this.getNumberOfDoubletonSuits() <= 1;
   }
 
+  public boolean hasEightOrMoreCardsInAnySuit() {
+    return hasNOrMoreCardsInLongestSuit(8);
+  }
+
+  public boolean hasSixCardsInLongestSuit() {
+    return hasNCardsInLongestSuit(6);
+  }
+
+  private boolean hasNCardsInLongestSuit(int n) {
+    int longestSuit = this.getNumberOfCardsInLongestSuit();
+    return longestSuit == n;
+  }
+
+  private boolean hasNOrMoreCardsInLongestSuit(int n) {
+    int longestSuit = this.getNumberOfCardsInLongestSuit();
+    return longestSuit >= n;
+  }
+
+  public boolean hasTwoOutOfThreeHigherCards() {
+    Predicate<Card> isThreeHigherCards = card -> card.getRank().equals(Rank.ACE) || card.getRank().equals(Rank.KING)
+        || card.getRank().equals(Rank.QUEEN);
+    long count = this.getCardsOfLongestSuit().stream().filter(isThreeHigherCards).count();
+    return count >= 2;
+  }
+
+  private List<Card> getCardsOfLongestSuit() {
+    Suit longestSuit = this.getLongestSuit();
+    Predicate<Card> isLongestSuit = card -> card.getSuit().equals(longestSuit);
+    return cards.stream().filter(isLongestSuit).collect(Collectors.toList());
+  }
+
+  private Suit getLongestSuit() {
```

Comment:
This should be enough to refactor this implementation:
```java
private Suit getLongestSuit(){
    return Collections.max(this.getCardsPerSuit().entrySet(), Comparator.comparingInt(Map.Entry::getValue)).getKey();
}
```
Anyway, what is the desired behavior when there is 2 different longest suits?